public/js/reducers/sources.js

```diff
@@ -156,24 +171,30 @@ function updateTabList(state, source, tabIndex) {
  * @memberof reducers/sources
  * @static
  */
-function getNewSelectedSourceId(state, id) : ?Source {
+function getNewSelectedSourceId(state, id: any) : ?Source {
```

Comment:
Yay! figured out how to push. @bomsy let me know what you think of the updates and if there's anything else we should do here.